app/services/order_adjustments_fetcher.rb

```diff
@@ -0,0 +1,64 @@
+class OrderAdjustmentsFetcher
```

Comment:
this needs a class level comment or a comment somewhere explaining the reason behind using .loaded?